src/main/java/org/jabref/logic/importer/fileformat/PdfContentImporter.java

```diff
@@ -218,280 +216,294 @@ public ParserResult importDatabase(Path filePath, Charset defaultEncoding) {
-            // i points to the current line
-            // curString (mostly) contains the current block
-            //   the different lines are joined into one and thereby separated by " "
-            lines = firstPageContents.split(System.lineSeparator());
-
-            lineIndex = 0; //to prevent array index out of bounds exception on second run we need to reset i to zero
-
-            proceedToNextNonEmptyLine();
-            if (lineIndex >= lines.length) {
-                // PDF could not be parsed or is empty
-                // return empty list
-                return new ParserResult();
-            }
+            Optional<BibEntry> entry = getEntryFromPDFContent(firstPageContents, OS.NEWLINE);
+            entry.ifPresent(result::add);
 
-            // we start at the current line
-            curString = lines[lineIndex];
-            // i might get incremented later and curString modified, too
-            lineIndex = lineIndex + 1;
-
-            String author;
-            String editor = null;
-            String abstractT = null;
-            String keywords = null;
-            String title;
-            String conference = null;
-            String DOI = null;
-            String series = null;
-            String volume = null;
-            String number = null;
-            String pages = null;
-            // year is a class variable as the method extractYear() uses it;
-            String publisher = null;
-
-            EntryType type = BibtexEntryTypes.INPROCEEDINGS;
-            if (curString.length() > 4) {
-                // special case: possibly conference as first line on the page
-                extractYear();
-                if (curString.contains("Conference")) {
+        } catch (EncryptedPdfsNotSupportedException e) {
+            return ParserResult.fromErrorMessage(Localization.lang("Decryption not supported."));
+        } catch (IOException exception) {
+            return ParserResult.fromError(exception);
+        } catch (FetcherException e) {
+            return ParserResult.fromErrorMessage(e.getMessage());
+        }
+
+        result.forEach(entry -> entry.addFile(new LinkedFile("", filePath.toAbsolutePath().toString(), "PDF")));
+        return new ParserResult(result);
+    }
+
+    //make this method package visible so we can test it
+    Optional<BibEntry> getEntryFromPDFContent(String firstpageContents, String lineSeparator) {
+
+        // idea: split[] contains the different lines
+        // blocks are separated by empty lines
+        // treat each block
+        //   or do special treatment at authors (which are not broken)
+        //   therefore, we do a line-based and not a block-based splitting
+        // i points to the current line
+        // curString (mostly) contains the current block
+        //   the different lines are joined into one and thereby separated by " "
+        lines = firstpageContents.split(lineSeparator);
```

Comment:
Since this reads the content of the pdf file, you cannot expect the newlines to be of a particular form. Use something like https://github.com/JabRef/jabref/blob/651b5454f12dc70d5cc1017b0d5acb817762f7ca/src/main/java/org/jabref/model/strings/StringUtil.java#L406

Reply:
Added this variant, merging now